File: src/main/java/net/yakclient/mixin/internal/methodadapter/tree/TreeOpcodePatternMatcher.java
package net.yakclient.mixin.internal.methodadapter.tree;

import net.yakclient.mixin.internal.instruction.ASMInstruction;
import net.yakclient.mixin.internal.instruction.Instruction;
import org.objectweb.asm.tree.InsnList;

public class TreeOpcodePatternMatcher extends TreeMixinPatternMatcher {
    private final int opcode;

    public TreeOpcodePatternMatcher(Instruction instructions, int opcode) {
        super(instructions);
        this.opcode = opcode;
    }

    @Override
    public void transform(InsnList insn) {

    }
}

File: base/src/main/kotlin/net/yakclient/mixins/base/internal/methodadapter/tree/TreeOpcodePatternMatcher.kt
package net.yakclient.mixins.base.internal.methodadapter.tree

import net.yakclient.mixins.base.internal.instruction.Instruction
import org.objectweb.asm.tree.InsnList

/**
 * Defines the pattern for inserting mixins before the opcode specified.
 *
 *
 * Type: Before Opcode
 *
 * @author Durgan McBroom
 */
class TreeOpcodePatternMatcher(instructions: Instruction?, private val opcode: Int) :
    TreeMixinPatternMatcher(instructions) {
    override fun transform(insn: InsnList) {
        for (node in insn) {
            if (node.opcode == opcode) this.insertInsn(node, insn)
        }
    }
}